lua/pac3/core/client/base_movable.lua

```diff
@@ -184,6 +189,58 @@ function PART:CalcAngles(ang, wpos)
 
 		return self.Angles + (pac.EyePos - wpos):Angle()
 	end
+	
+	if pac.StringFind(self.AimPartName, "NEAREST_LIFE_YAW", true, true) then
+		local nearest_ent = self:GetRootPart():GetOwner()
+		local nearest_dist = math.huge
+		local owner_ent = self:GetRootPart():GetOwner()
+		local part_ent = self:GetOwner()
+		for _,ent in pairs(ents.GetAll()) do
+			if (ent:IsNPC() or ent:IsPlayer()) and ent ~= owner_ent then
+				local dist = (owner_ent:GetPos() - ent:GetPos()):LengthSqr()
+				if dist < nearest_dist then
+					nearest_ent = ent
+					nearest_dist = dist
+				end
+			end
+		end
+		local ang = (nearest_ent:GetPos() - part_ent:GetPos()):Angle()
+		return Angle(0,ang.y,0) + self.Angles
+	end
+	
+	if pac.StringFind(self.AimPartName, "NEAREST_LIFE_POS", true, true) then
+		local nearest_ent = self:GetRootPart():GetOwner()
+		local nearest_dist = math.huge
+		local owner_ent = self:GetRootPart():GetOwner()
+		local part_ent = self:GetOwner()
+		for _,ent in pairs(ents.GetAll()) do
+			if (ent:IsNPC() or ent:IsPlayer()) and ent ~= owner_ent then
+				local dist = (owner_ent:GetPos() - ent:GetPos()):LengthSqr()
+				if dist < nearest_dist then
+					nearest_ent = ent
+					nearest_dist = dist
+				end
+			end
+		end
+		return self.Angles + (nearest_ent:GetPos() - part_ent:GetPos()):Angle()
+	end
+	
+	if pac.StringFind(self.AimPartName, "NEAREST_LIFE", true, true) then
+		local nearest_ent = self:GetRootPart():GetOwner()
+		local nearest_dist = math.huge
+		local owner_ent = self:GetRootPart():GetOwner()
+		local part_ent = self:GetOwner()
+		for _,ent in pairs(ents.GetAll()) do
+			if (ent:IsNPC() or ent:IsPlayer()) and ent ~= owner_ent then
+				local dist = (owner_ent:GetPos() - ent:GetPos()):LengthSqr()
+				if dist < nearest_dist then
+					nearest_ent = ent
```

Comment:
can you create a local helper function that gets all of the nearby entities to avoid the repetition?

Reply:
for readability or for performance?

Reply:
> gets all of the nearby entities
That's... just ents.findinsphere or findinbox. it already exists. although I used ents.getall in this case. But perhaps it should pick the sphere one and for it to be limited to a certain radius. I'll pick 5000 or something.

The only repetition is the processing that comes after, to get the nearest entity. I'll compact the loops into that helper function you wanted.